src/main/check.cpp

```diff
@@ -182,5 +182,10 @@ int main(int argc, char** argv) {
     } else {
         assert(false);
     }
+
+    // Call clear thread local state to ensure that the verifier does not
+    // leak memory.
+    ebpf_verifier_clear_thread_local_state();
```

Comment:
We'll have a stronger guarantee if it is inside a try-catch, or maybe better, as a destructor of a scoped variable. Just a thought.